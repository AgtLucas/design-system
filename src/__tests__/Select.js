import React from 'react'
import renderer from 'react-test-renderer'
import { Select } from '..'

describe('Select', () => {
  test('renders', () => {
    const json = renderer.create(<Select>Test</Select>).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with isExpanded prop', () => {
    const json = renderer
      .create(<Select isExpanded>Test with isExpanded</Select>)
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
