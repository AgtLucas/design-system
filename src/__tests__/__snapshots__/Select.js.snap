// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders 1`] = `
.c1 {
  text-align: center;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: blue;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: black;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  height: 48px;
  border-width: 1px;
  border-style: solid;
}

.c2 {
  padding-left: 32px;
  cursor: pointer;
}

.c4 {
  padding-right: 32px;
  cursor: pointer;
}

.c6 {
  padding-left: 5px;
  padding-right: 32px;
  cursor: pointer;
}

.c5 {
  font-size: 10px;
  display: block;
  width: 100%;
  margin-left: 16px;
  font-size: 14px;
  text-align: left;
  cursor: pointer;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
  >
    <svg
      className="c3"
      color="blue"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12.6,10.1C11.8,8,9.8,6.5,7.5,6.5C4.4,6.5,2,8.9,2,12s2.4,5.5,5.5,5.5c2.4,0,4.4-1.5,5.1-3.6h4
      	v3.6h3.6v-3.7H22v-3.7H12.6z M7.5,13.8c-1,0-1.8-0.8-1.8-1.8s0.8-1.8,1.8-1.8S9.3,11,9.3,12S8.5,13.8,7.5,13.8z"
      />
    </svg>
  </div>
  <div
    className="c4"
  >
    <label
      className="c5"
      fontSize={1}
    >
      Test
    </label>
  </div>
  <div
    className="c6"
  >
    <svg
      className="c7"
      color="black"
      fill="currentcolor"
      height={12}
      viewBox="0 0 20 11"
      width={12}
    >
      <path
        d="M19.998 1.074a.995.995 0 0 0-.437-.836 1.21 1.21 0 0 0-.76-.234 1.113 1.113 0 0 0-.784.368l-7.783 8L1.917.308C1.49-.102.744-.102.319.306.116.5 0 .78 0 1.074c0 .293.116.571.32.767l9.13 8.85c.204.196.495.309.8.309.34 0 .643-.14.843-.387L19.69 1.83c.197-.194.31-.47.31-.757h-.002z"
      />
    </svg>
  </div>
</div>
`;

exports[`Select renders with isExpanded prop 1`] = `
.c1 {
  text-align: center;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: blue;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  height: 48px;
  border-width: 1px;
  border-style: solid;
}

.c2 {
  padding-left: 32px;
  cursor: pointer;
}

.c4 {
  padding-right: 32px;
  cursor: pointer;
}

.c6 {
  padding-left: 5px;
  padding-right: 32px;
  cursor: pointer;
}

.c5 {
  font-size: 10px;
  display: block;
  width: 100%;
  margin-left: 16px;
  font-size: 14px;
  text-align: left;
  cursor: pointer;
}

.c7 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  color: black;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

<div
  className="c0 c1"
>
  <div
    className="c2"
  >
    <svg
      className="c3"
      color="blue"
      fill="currentcolor"
      height={24}
      viewBox="0 0 24 24"
      width={24}
    >
      <path
        d="M12.6,10.1C11.8,8,9.8,6.5,7.5,6.5C4.4,6.5,2,8.9,2,12s2.4,5.5,5.5,5.5c2.4,0,4.4-1.5,5.1-3.6h4
      	v3.6h3.6v-3.7H22v-3.7H12.6z M7.5,13.8c-1,0-1.8-0.8-1.8-1.8s0.8-1.8,1.8-1.8S9.3,11,9.3,12S8.5,13.8,7.5,13.8z"
      />
    </svg>
  </div>
  <div
    className="c4"
  >
    <label
      className="c5"
      fontSize={1}
    >
      Test with isExpanded
    </label>
  </div>
  <div
    className="c6"
  >
    <svg
      className="c7"
      color="black"
      fill="currentcolor"
      height={12}
      viewBox="0 0 20 11"
      width={12}
    >
      <path
        d="M19.998 1.074a.995.995 0 0 0-.437-.836 1.21 1.21 0 0 0-.76-.234 1.113 1.113 0 0 0-.784.368l-7.783 8L1.917.308C1.49-.102.744-.102.319.306.116.5 0 .78 0 1.074c0 .293.116.571.32.767l9.13 8.85c.204.196.495.309.8.309.34 0 .643-.14.843-.387L19.69 1.83c.197-.194.31-.47.31-.757h-.002z"
      />
    </svg>
  </div>
</div>
`;
